omod/src/main/java/org/openmrs/module/aijar/page/controller/MarkPatientDeadPageController.java

```diff
@@ -41,28 +44,28 @@ public void get(@SpringBean PageModel pageModel, @RequestParam(value = "breadcru
         }
     }
 
-    public String post(@RequestParam(value = "causeOfDeath",required = false) String causeOfDeath, @RequestParam(value = "dead",required = false) Boolean dead, @RequestParam(value = "deathDate",required = false) Date deathDate, @RequestParam("patientId") String patientId, UiUtils
-            uiUtils) {
-        PersonService personService = Context.getPersonService();
+    public String post(@RequestParam(value = "causeOfDeath", required = false) String causeOfDeath, @RequestParam(value = "dead", required = false) Boolean dead, @RequestParam(value = "deathDate", required = false) Date deathDate, @RequestParam("patientId") String patientId) {
+
+        PatientService patientService = Context.getPatientService();
+        Patient patient = patientService.getPatientByUuid(patientId);
         try {
-            PatientService patientService = Context.getPatientService();
-            Patient patient = patientService.getPatientByUuid(patientId);
-            Person person = personService.getPerson(patient.getPatientId());
-            if (dead!=null && causeOfDeath != null && deathDate != null) {
-                person.setDead(dead);
-                person.setCauseOfDeath(getConceptByUUId(causeOfDeath));
-                person.setDeathDate(deathDate);
+            Date date = new Date();
+
+            if (dead != null && !causeOfDeath.equals("") && deathDate != null && !deathDate.before(patient.getBirthdate()) && !deathDate.after(date)) {
+                patient.setDead(dead);
+                patient.setCauseOfDeath(getConceptByUUId(causeOfDeath));
+                patient.setDeathDate(deathDate);
             } else {
-                person.setDeathDate(null);
-                person.setDead(false);
-                person.setCauseOfDeath(null);
+                patient.setDeathDate(null);
+                patient.setDead(false);
+                patient.setCauseOfDeath(null);
             }
 
-            personService.savePerson(person);
+            patientService.savePatient(patient);
             return "redirect:/coreapps/clinicianfacing/patient.page?patientId=" + patient.getUuid() + "";
         } catch (Exception e) {
-            System.out.println(e);
-            return null;
+            log.error(e.getMessage());
+                return "redirect:/aijar/markPatientDead.page?patientId={{" + patient.getUuid() + "}}";
```

Comment:
@slubwama Remove the {{}} which are used as placeholders yet you are passing the actual value